File: my-folder/0189-rotate-array/solution.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        k = k % len(nums)
        nums1=nums[-k:] 
        nums2= nums[:-k]
        nums[:]=nums1+nums2

File: my-folder/0189-rotate-array/solution.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n = nums.size();
        k = k % n;
        
        vector<int> nums1(nums.begin() + (n - k), nums.end());
        vector<int> nums2(nums.begin(), nums.begin() + (n - k));
        
        nums = nums1;
        nums.insert(nums.end(), nums2.begin(), nums2.end());
    }
};
